concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState/Updates.hs

```diff
@@ -716,7 +740,12 @@ migrateUpdates migration Updates{..} = do
     newParameters <- migrateHashedBufferedRef (return . StoreSerialized . migrateChainParameters migration . unStoreSerialized) currentParameters
     newCurrentProtocolUpdate <- case currentProtocolUpdate of
         Null -> return Null
-        Some c -> Some <$!> migrateHashedBufferedRefKeepHash c
+        -- For P5->P6 we clear the currentProtocolUpdate as part of the migration.
+        -- For older versions the @currentProtocolUpdate@ is cleared before i.e. as
+        -- part of 'updateRegenesis'.
+        Some c -> case (protocolVersion @pv) of
```

Comment:
You could probably just clear it always. But I haven't checked if we actually rely on it anywhere later. I don't think we do, but this is also fine.